source/GameAction.cpp

```diff
@@ -392,9 +410,21 @@ GameAction GameAction::Instantiate(map<string, string> &subs, int jumps, int pay
 
 	for(auto &&it : giftShips)
 		result.giftShips.emplace_back(it.first, !it.second.empty() ? it.second : GameData::Phrases().Get("civilian")->Get());
-	result.giftOutfits = giftOutfits;
 
-	result.payment = payment + (jumps + 1) * payload * paymentMultiplier;
+	// Add any additional required outfits
+	map<const Outfit *, int> mergedGiftedOutfits(giftOutfits);
+	int outfitObjectiveCost = 0;
+	for(auto &it : outfitObjective)
+	{
+		outfitObjectiveCost += it.first->Cost() * it.second * OutfitBulkBonus(it.second);
+
+		if(mergedGiftedOutfits.count(it.first) == 0)
+			mergedGiftedOutfits[it.first] = 0;
```

Comment:
Whats the idea here? We shouldnt have an outfit in it with it being 0 I think?

Reply:
If the key isn't present initialize it at 0, then add the value to the new merged value.

Reply:
Change line 421 by `if(!mergedGiftedOutfits.count(it.first))` that'll make it easier to read I think